File: src/nl/hannahsten/texifyidea/psi/LatexStubElementTypes.java
package nl.hannahsten.texifyidea.psi;

import nl.hannahsten.texifyidea.index.stub.LatexCommandsStubElementType;
import nl.hannahsten.texifyidea.index.stub.LatexEnvironmentStubElementType;
import nl.hannahsten.texifyidea.index.stub.LatexMagicCommentStubElementType;

/**
 * Grammar-Kit cannot generate this file containing only stub element types, so we have to maintain it manually.
 * This file has to be in Java, otherwise it will not be recognised as lazily loadable by IStubElementType#checkNotInstantiatedTooLateWithId
 */
public interface LatexStubElementTypes {

    LatexCommandsStubElementType COMMANDS = (LatexCommandsStubElementType) LatexTypes.COMMANDS;
    LatexEnvironmentStubElementType ENVIRONMENT = (LatexEnvironmentStubElementType) LatexTypes.ENVIRONMENT;
    LatexMagicCommentStubElementType MAGIC_COMMENT = (LatexMagicCommentStubElementType) LatexTypes.MAGIC_COMMENT;
}

File: src/nl/hannahsten/texifyidea/psi/LatexStubElementTypes.kt
package nl.hannahsten.texifyidea.psi

import nl.hannahsten.texifyidea.grammar.LatexStubFileElementType
import nl.hannahsten.texifyidea.index.stub.LatexCommandsStubElementType
import nl.hannahsten.texifyidea.index.stub.LatexEnvironmentStubElementType
import nl.hannahsten.texifyidea.index.stub.LatexMagicCommentStubElementType

/**
 * Grammar-Kit cannot generate this file containing only stub element types, so we have to maintain it manually.
 */
@Suppress("unused")
interface LatexStubElementTypes {

    companion object {

        @JvmField
        val COMMANDS = LatexTypes.COMMANDS as LatexCommandsStubElementType

        @JvmField
        val ENVIRONMENT = LatexTypes.ENVIRONMENT as LatexEnvironmentStubElementType

        @JvmField
        val MAGIC_COMMENT = LatexTypes.MAGIC_COMMENT as LatexMagicCommentStubElementType

        @JvmField
        val FILE = LatexStubFileElementType
    }
}